databases/week1/homework1.sql

```diff
@@ -0,0 +1,92 @@
+USE HYF_lesson1
+-- Find out how many tasks are in the task table
+SELECT
+  COUNT(id)
+FROM
+  task;
+
+-- Find out how many tasks in the task table do not have a valid due date
+SELECT
+  count(id)
+FROM
+  task
+where
+  due_date IS NULL;
+
+-- Find all the tasks that are marked as done
+SELECT
+  *
+FROM
+  task
+WHERE
+  status_id = 3;
```

Comment:
What will happen if we change the records in the `status` table and "Done" get id `2`? 
The idea here is to use `JOIN` between tables. In the case above, the specific `id` doesn't matter 

```suggestion
SELECT task.id, task.name, status.name
FROM task 
   JOIN status on task.status_id = status.id
WHERE status.name = "Done";
```

Reply:
Didn't get the meaning from the text of assignment.

Reply:
When the records have been created in `status` table we don't (or we shouldn't) know which status name has which id, i.e. `done` may be any id 

| id | name |
-----------
| 1  | Done  |
| 2  | Ready |
| 3  | In progress |

So, In this case, the query will return all tasks in progress.

We will discuss this in the class